### PR TITLE
API Description reader class with schema validation

### PR DESCRIPTION
**Pull request by automatthew**

### CODE REVIEW

File: schema/api-description.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      signature: {$ref: "#/definitions/signature"}
+
+  signature:
+    type: object
+    properties:
+      request:
+        type: string
+      response:
+        type: string
+      status:
```
Comment (dyoder) [inline schema/api-description.yaml]: Doesn't status go under response?

File: src/api.coffee [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+validator = Schemas.validator "api-description"
+
+module.exports = class API
+
+  @read: async (apiPath) ->
+    # TODO: allow either a yaml file or a directory of yaml files
+    new @ yaml yield read apiPath
+
+  constructor: (description) ->
+    {valid, errors} = validator.validate description
```
Comment (dyoder) [inline src/api.coffee]: yay

File: src/commands/update.coffee [hunk 3]
```diff
@@ -2,14 +2,17 @@
...
+{bellChar} = require "../utils"
+configuration = require "../configuration"
 
 
 define "update", async (env) ->
   try
-    config = yield require("./configuration/compile")(env)
-    {lambdaUpdate} = yield require("./aws/app-root")(env, config)
-    api = yaml yield read join process.cwd(), "api.yaml"
+    appRoot = process.cwd()
```
Comment (dyoder) [inline src/commands/update.coffee]: more `appRoot` shenanigans

File: src/configuration/cloudformation.coffee [hunk 4]
```diff
@@ -34,9 +27,12 @@ renderTemplate = async (appRoot, globals) ->
...
   }
 
-# Finds and renders all mixins as the Resources for a CloudFormation Template.
+# Finds and renders the API description and all mixins as the Resources for a
+# CloudFormation Template.
 renderResources = async (appRoot, globals) ->
   resources = []
+  resources.push yield renderAPI appRoot, globals
+
   mixinNames = yield listMixins appRoot
```
Comment (dyoder) [inline src/configuration/cloudformation.coffee]: step in the right direction :)

File: schema/api-description.yaml (hunk 1 above)
Comment (automatthew) [inline schema/api-description.yaml]: This is for the existing format, as used in Blurb9 today.

**Pull request merged**